src/accelerate/utils/dataclasses.py

```diff
@@ -742,7 +742,7 @@ def set_deepspeed_weakref(self):
                 or ds_config["train_micro_batch_size_per_gpu"] == "auto"
             ):
                 ds_config["train_micro_batch_size_per_gpu"] = 1
-            if ds_config["train_batch_size"] == "auto":
+            if "train_batch_size" in ds_config and ds_config["train_batch_size"] == "auto":
                 del ds_config["train_batch_size"]
```

Comment:
```suggestion
            if ds_config.get("train_batch_size", None) == "auto":
                del ds_config["train_batch_size"]
```

Let's make it a bit cleaner :)